src/bigcommerce.js

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable eslint-comments/no-unlimited-disable */
+/* eslint-disable */
+
+if (window.SGJavascriptBridge) {
+  shopgateInit();
+} else if (navigator.userAgent.indexOf('libshopgate') !== -1) { // Shopgate iOS App
+  function shopgateExecuteWithRetry(
+    intervalInMiliseconds,
+    maximumIntervallTimeInMiliseconds,
+    executeCallback
+  ) {
+    const startTimestampInMiliseconds = Date.now();
+
+    const interval = setInterval(() => {
+      if (startTimestampInMiliseconds + maximumIntervallTimeInMiliseconds <= Date.now()) {
+        clearInterval(interval);
+        return;
+      }
+
+      if (!executeCallback()) {
+        return;
+      }
+
+      clearInterval(interval);
+    }, intervalInMiliseconds);
+  }
+
+  shopgateExecuteWithRetry(25, 2500, function() {
+    if (!window.SGJavascriptBridge) {
+      return false;
+    }
+
+    shopgateInit();
+
+    return true;
+  });
+}
+
+/**
+ * this will only be executed when a Shopgate App visits the desktop site.
+ */
+function shopgateInit() {
+  var shopgateShopNumber = 'REPLACE_WITH_SHOP_NUMBER';
```

Comment:
Would it be fine for you if assignment was at the very top for having it easier to whomever does the integration? I know it would be exposed even if the page is not loaded through our app, I'm just wondering what you think about it.

Reply:
rearranged methods inside of bigcommerce.js to make it easier to adapt and understand